src/main/java/de/tum/in/www1/artemis/service/VcsAccessLogService.java

```diff
@@ -0,0 +1,86 @@
+package de.tum.in.www1.artemis.service;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Repository;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.participation.Participation;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.domain.vcstokens.AuthenticationMechanism;
+import de.tum.in.www1.artemis.domain.vcstokens.VcsAccessLog;
+import de.tum.in.www1.artemis.repository.ParticipationRepository;
+import de.tum.in.www1.artemis.repository.VcsAccessLogRepository;
+import de.tum.in.www1.artemis.web.rest.repository.RepositoryActionType;
+
+@Profile(PROFILE_CORE)
+@Service
+public class VcsAccessLogService {
```

Comment:
The title of the PR suggests that we only want to have an access log for LocalVC repositories. On TS6 (with a GitLab setup), I was able to write an entry into the table using the code editor:

```
GET https://artemis-test6.artemis.cit.tum.de/api/programming-exercise-participations/45/vcs-access-log
[
    {
        "id": 1,
        "userId": 21,
        "name": "Patrik Zander",
        "email": "patrik.zander@tum.de",
        "repositoryActionType": "WRITE",
        "authenticationMechanism": "CODE_EDITOR",
        "commitHash": "abf958cf5d48c548d054ed4336b52357e6cb60a2",
        "timestamp": "2024-08-28T07:56:33.808948Z"
    }
]
```

Is this intended?

Reply:
Hmm, that's actually a good question. As I understand we want to generally move from gitlab to the localVC setup, and having the access log for the code editor interactions don't really have any impact. 
Only thing to note is, that the access log would only contain the code editor interactions. I see three options here:
1. Ignore that the log is written for gitlab code editor pushs, and let instructors see the log (although its incomplete)
2. Ignore that the log is written for gitlab code editor pushs, but disable the access log view (Data is written but can't be accessed)
3. Add a check, and don't add logs if gitlab is used + disable the access log views
4. Find a way to also log push operations to gitlab. -> Not sure if its worth to invest this effort, especially as that functionality is most likely already in gitlab, and, we want to move away from gitlab.